Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,52 +2,84 @@
+ * @param useGlobalJson A switch so we know if the user have `global.json` files and want use that.
+ * @param workingDirectory This is only relevant if the `useGlobalJson` switch is `true`. It will set the root directory for the search of `global.json`
+ * @param includePreviewVersions Define if the installer also search for preview version
+ * @param performMultiLevelLookup Set the `DOTNET_MULTILEVEL_LOOKUP`environment variable
+ */
+async function installDotNet(
+    installationPath: string,
+    packageType: string,
+    versionSpec: string | null,
+    useGlobalJson: boolean,
+    workingDirectory: string | null,
+    includePreviewVersions: boolean,
+    performMultiLevelLookup: boolean) {
 
-        let versionFetcher = new DotNetCoreVersionFetcher();
+    if (!installationPath && installationPath.length == 0) {
+        installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
+    }
+    let versionFetcher = new DotNetCoreVersionFetcher();
+    let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
+    if (useGlobalJson) {
+        let globalJsonFetcherInstance = new globalJsonFetcher(workingDirectory);
+        let versionsToInstall: VersionInfo[] = await globalJsonFetcherInstance.Get(packageType);        
+        versionsToInstall = versionsToInstall.filter(d => !dotNetCoreInstaller.isVersionInstalled(d.getVersion()));
+        for (let index = 0; index < versionsToInstall.length; index++) {
+            const version = versionsToInstall[index];
+            let url = versionFetcher.getDownloadUrl(version);
+            await dotNetCoreInstaller.downloadAndInstall(version, url);            
+        }
+    } else if (versionSpec) {
+        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+        let versionSpecParts = new VersionParts(versionSpec);        
         let versionInfo: VersionInfo = await versionFetcher.getVersionInfo(versionSpecParts.versionSpec, packageType, includePreviewVersions);
         if (!versionInfo) {
             throw tl.loc("MatchingVersionNotFound", versionSpecParts.versionSpec);
         }
-
-        let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
         if (!dotNetCoreInstaller.isVersionInstalled(versionInfo.getVersion())) {
             await dotNetCoreInstaller.downloadAndInstall(versionInfo, versionFetcher.getDownloadUrl(versionInfo));
         }
-
-        tl.prependPath(installationPath);
-
-        // Set DOTNET_ROOT for dotnet core Apphost to find runtime since it is installed to a non well-known location.
-        tl.setVariable('DOTNET_ROOT', installationPath);
-
-        // By default disable Multi Level Lookup unless user wants it enabled.
-        let performMultiLevelLookup = tl.getBoolInput("performMultiLevelLookup", false);
-        tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");
+    }else{
+        throw tl.loc("installDotNetCalledWithWrongParameters")
```

Comment:
The current behavior of the task is not throw if there is nothing to install. You should maintain this

Reply:
No that is not the case. This error only is raised if a program of this task does a mistake. You can't raise this error if you use the task. That error should make debugging easier.